src/util/invariant.h

```diff
@@ -160,10 +162,17 @@ invariant_violated_structured(
   const std::string &file,
   const std::string &function,
   const int line,
+  const std::string &condition,
   Params &&... params)
 {
   std::string backtrace=get_backtrace();
-  ET to_throw(file, function, line, backtrace, std::forward<Params>(params)...);
+  ET to_throw(
+    file,
+    function,
+    line,
+    backtrace,
+    std::forward<Params>(params)...,
```

Comment:
Can we have the variable number of arguments as the last argument? (i.e., change the order of this line and the subsequent one). We also need to change the order of the `reason` and the `condition` parameters of the constructor of `invariant_failedt` then.